Architecture.md

```diff
@@ -0,0 +1,85 @@
+# Architecture
+
+This is just a proposal of a potential set of classes / services and their responsibilities and what APIs they may have.
+
+## Top-Level Objects
+
+<table>
+  <thead>
+    <tr><th style='width: 120px'>Object</th><th>Description / Responsibility</th><th>Example APIs</th>
+  </thead>
+  <tbody>
+    <tr>
+      <td>Model</td><td>defines the attributes and relationships for resources </td>
+      <td>
+        - save()<br>
+        - destroy()<br>
+        - delete()<br>
+        - rollbackAttributes()<br>
+        - hasDirtyAttributes()<br>
+        - errors<br>
+        - serialize()<br>
+      </td>
+    </tr>
+    <tr>
+      <td>Store</td>
+      <td>
+        - retrieves models from the locale storage<br>
+        - retrieves models from the server<br>
+        - performs CRUD on any resource
+      </td>
+      <td>
+        - createRecord(type, params)<br>
+        - findRecord(type, options)<br>
+        - peekRecord(type, id)<br>
+        - findAll(type)<br>
+        - peekAll(type)<br>
+        - query(type, params)<br>
+        - queryRecord(type, params)<br>
```

Comment:
We might want to add a method to sync an arbitrary payload into the store (for testing for instance).